### PR TITLE
[RDY]vim-patch:8.0.1554,8.1.1977,8.2.{927,930,932,938,954,963,964,966,980,983,998,999}

### PR DESCRIPTION
**Pull request by janlazo**

### CODE REVIEW

File: src/nvim/option.c [hunk 1]
```diff
@@ -524,11 +524,15 @@ char *get_lib_dir(void)
...
+///
+/// If "clean_arg" is true, Nvim was started with --clean.
+static void set_runtimepath_default(bool clean_arg)
 {
   size_t rtp_size = 0;
   char *const data_home = stdpaths_get_xdg_var(kXDGDataHome);
-  char *const config_home = stdpaths_get_xdg_var(kXDGConfigHome);
+  char *const config_home = clean_arg
+    ? NULL
+    : stdpaths_get_xdg_var(kXDGConfigHome);
```
Comment (janlazo) [inline src/nvim/option.c:535]: This removes `stdpath('config')` from `&runtimepath`. Not sure if `stdpath('data')` and other directories should be excluded as well.

File: runtime/doc/starting.txt [hunk 2]
```diff
@@ -90,6 +90,7 @@ argument.
 --clean		Equivalent to "-u NONE -i NONE":
 		- Skips initializations from files and environment variables.
 		- No 'shada' file is read or written.
+		- Excludes home directory from 'runtimepath'
```
Comment (janlazo) [inline runtime/doc/starting.txt:93]: Does this mean `stdpath('config')` directory only?

File: src/nvim/option.c (hunk 1 above)
Comment (jamessan) [reply src/nvim/option.c:535]: Yeah, all of the user-specific config should be ignored.  Only the system-wide config should be available.

**Comment by teto** [comment]
went through it and it looked perfect. Thanks for the PVS too. I will be diligent on the next version.c patch, please repost your NA patches ^^''

**Pull request merged**